### PR TITLE
Add nonlinear iterations for FSI

### PR DESCRIPTION
**Pull request by gantech**

### CODE REVIEW

**Review by psakievich — APPROVED** [review]

**Comment by ashesh2512** [comment]
Don't merge before #46 - we need to make sure that the timing of certain calls is incremented.

File: src/OversetSimulation.cpp
Comment (ashesh2512) [inline src/OversetSimulation.cpp:151]: change to - 
`increment_timer = inonlin > 0 ? true : false;`
`exchange_solution(increment_timer);`

Comment (ashesh2512) [inline src/OversetSimulation.cpp:151]: exchange_solution(increment_timer);

Comment (ashesh2512) [inline src/OversetSimulation.cpp:151]: change to 
`increment_timer = inonlin > 0 ? true : false`
`exchange_solution(increment_timer)`

Comment (ashesh2512) [reply src/OversetSimulation.cpp:151]: Once this PR is merged, I will add the timer increments to other methods under the nonlinear iteration loop.

**Review by marchdf — APPROVED** [review]

**Review by ashesh2512 — APPROVED** [review]

**Pull request merged**
